src/common/functions-v2/tests/it/scalars/string.rs

```diff
@@ -346,10 +346,32 @@ fn test_trim(file: &mut impl Write) {
     test_trim_with_from(file, "trailing");
 }
 
-fn build_string_column(strings: &[&str]) -> Column {
-    let mut builder = ColumnBuilder::with_capacity(&DataType::String, strings.len());
-    for s in strings {
-        builder.push(ScalarRef::String(s.as_bytes()));
-    }
-    builder.build()
+fn test_concat(file: &mut impl Write) {
+    run_ast(file, "concat('5', '3', '4')", &[]);
+    run_ast(file, "concat(NULL, '3', '4')", &[]);
+    run_ast(file, "concat(a, '3')", &[(
+        "a",
+        DataType::String,
+        Column::from_data(&["abc", "   abc", "   abc   ", "abc   "]),
+    )]);
+
+    run_ast(file, "concat(a, '3')", &[(
+        "a",
+        DataType::Nullable(Box::new(DataType::String)),
+        Column::from_data_with_validity(&["a", "b", "c", "d"], vec![true, true, false, true]),
+    )]);
+
+    run_ast(file, "concat_ws('-', '3', '4')", &[]);
+    run_ast(file, "concat_ws(NULL, '3', '4')", &[]);
+    run_ast(file, "concat_ws(a, '3')", &[(
```

Comment:
let's test a little bit more

```suggestion
    run_ast(file, "concat_ws(a, '3', '4')", &[(
```